src/store_client.cc

```diff
@@ -564,28 +567,47 @@ store_client::unpackHeader(char const *buf, ssize_t len)
     assert(len >= 0);
 
     int swap_hdr_sz = 0;
-    tlv *tlv_list = nullptr;
+    SBuf varyHeaders;
     try {
+        using namespace Store;
         StoreMetaUnpacker aBuilder(buf, len, &swap_hdr_sz);
-        tlv_list = aBuilder.createStoreMeta();
-    } catch (const std::exception &e) {
-        debugs(90, DBG_IMPORTANT, "WARNING: failed to unpack metadata because " << e.what());
-        return false;
-    }
-    assert(tlv_list);
-
-    /*
-     * Check the meta data and make sure we got the right object.
-     */
-    for (tlv *t = tlv_list; t; t = t->next) {
-        if (!t->checkConsistency(entry)) {
-            storeSwapTLVFree(tlv_list);
-            return false;
+        for (const auto &meta: aBuilder) {
+            switch (meta.type) {
+            case STORE_META_VOID:
+                // aBuilder already complained if needed
```

Comment:
Polished in 72fbab4.

A comment is needed here to clarify why we are ignoring this never-serialized special value coming from a deserializer (e.g., instead of asserting or complaining about something that should not happen). Improvements are welcomed.

A "do nothing" comment would be worse that a comment explaining _why_ we are (obviously) doing nothing.

> IMO we should allow TLV entries to be updated to something like this value

If this plan ever materializes, we will use a different/new type value dedicated to that purpose. Today, a never-serialized `STORE_META_VOID` is needed to signal deprecated and other swap meta fields that the caller can ignore.

Same for https://github.com/squid-cache/squid/pull/1067/files#r889919746